caliban_toolbox/figure_eight_functions.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2016-2020 David Van Valen at California Institute of Technology
+# (Caltech), with support from the Paul Allen Family Foundation, Google,
+# & National Institutes of Health (NIH) under Grant U24CA224309-01.
+# All rights reserved.
+#
+# Licensed under a modified Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.github.com/vanvalenlab/caliban-toolbox/LICENSE
+#
+# The Work provided may be used for non-commercial academic purposes only.
+# For any other use of the Work, including commercial use, please contact:
+# vanvalenlab@gmail.com
+#
+# Neither the name of Caltech nor the names of its contributors may be used
+# to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import requests
+import os
+import stat
+import zipfile
+import pandas as pd
+
+from getpass import getpass
+from caliban_toolbox.log_file import create_upload_log
+from caliban_toolbox.aws_functions import aws_upload_files, aws_download_files
+
+
+def copy_job(job_id, key):
+    """Helper function to create a Figure 8 job based on existing job.
+
+    Args:
+        job_id: ID number of job to copy instructions and settings from when creating new job
+        key: API key to access Figure 8 account
+
+    Returns:
+        int: ID number of job created
+    """
+
+    url = 'https://api.appen.com/v1/jobs/{}/copy.json?'.format(str(job_id))
+    API_key = {"key": key}
+
+    new_job = requests.get(url, params=API_key)
+    if new_job.status_code != 200:
+        print("copy_job not successful. Status code: ", new_job.status_code)
+    new_job_id = new_job.json()['id']
+
+    return new_job_id
+
+
+def upload_data(csv_path, job_id, key):
+    """Add data to an existing Figure 8 job by uploading a CSV file
+
+    Args:
+        csv_path: full path to csv
+        job_id: ID number of job to upload data to
+        key: API key to access Figure 8 account
+    """
+
+    url = "https://api.appen.com/v1/jobs/{job_id}/upload.json?key={api_key}&force=true"
+    url = url.replace('{job_id}', str(job_id))
+    url = url.replace('{api_key}', key)
```

Comment:
`.format()` seems like a better option than the calling `replace()`.

Reply:
Agreed. I'm gonna look into url formatting for all of it. Seeing now if capitalization of true/false matters for caliban